README.md

```diff
@@ -1,2 +1,233 @@
+of concern. This means there's a simple design with minimal dependencies. As
+inspiration,
+see
+[Bittorrent economics paper (pdf)](http://users.eecs.northwestern.edu/~hq/papers/communities.pdf),
+which is a separate protocol layer that people can choose to use or not to get
+better quality of service (request and choking).
+
+### What about Message Reliability?
+The protocol should do Best Effort Delivery. Reliable Delivery can be provided
+on top, similar to TCP/IP (or BSP/BTP for Briar), to accommodate things like:
+
+- Guaranteed Message Delivery
+- Message Ordering
+- And possibly: Asynchronous Messaging (some protocols deal with it at this
+  layer, so this may or may not be desirable)
+
+Depending on the specifics of the reliability mechanism, throughput etc, this
+may have consequences for the above Reasonable Latency requirement. The
+Reasonable Latency requirement outlined above is for End to End messaging.
+
+### How to deal with Network Spam?
+- One approach is to use a Friend-to-Friend (F2F) network. This is what Briar
+  and Secure Scuttlebutt (SSB) does.
+- For open DHT-based, another approach is to rely on proof of work like Whisper.
+  This isn't very practical for mobile / resource restricted devices, and
+  appears to have limited usability.
+- More approaches are likely possibly, such as traditional rate limiting, basic
+  peer reputation, payments, etc.
+- Global network attacks more relevant here than a specific node. What does this
+  imply for a DHT?
+
+### How to deal with Asynchronous Messaging?
+- One approach is to punt this problem to data sync layer.
+- Another example is Briar requiring two entities to both be online
+- Not clear that it is a necessary component of AC layer.
+- One idea is to use Aggregation Points (Xolotl, lake mixnet) as providers
+  similar to Loopix, but presumably with less HA guarantees.
+
+## Initial work
+
+###  Network
+We would like to develop a protocol that leverages the best research and
+protocols from the past in order to create something for the future of
+decentralised applications. We want a “roadmap without potholes” for providing
+stronger privacy assurances than Tor for both senders and receivers, while also
+scaling well and providing short-term message storage for offline users. In
+other words, we accept that a project this large requires a piecemeal approach
+but we shall understand and avoid design dead ends that prevent either scaling
+or rigorous analysis of anonymity properties.
+
+We think mix networks occupy a sweet spot in which nodes run extremely efficiently but rigorous analysis remains possible, if challenging.  There are designs like Tor with more efficient designs, but they cannot provide rigorous anonymity properties.  There are also numerous academic schemes designed to support rigorous analysis, but at extreme sacrifices in efficiency.  We choose a Sphinx-like packet format because it’s efficient, adaptable enough, and has good security proofs.  
```

Comment:
Might be useful to more explicitly contrast mix network with other approaches, such as broadcast based, onion routing, (PIR/DC) so it captures Whisper/PSS/Tor issues.

Reply:
Yes, ideally we have a clear motivation for the direction we end up going in, can be done later though.